presto-spi/src/main/java/io/prestosql/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -474,19 +474,24 @@ public static void multiply(Slice left, Slice right, Slice result)
     {
         checkArgument(result.length() == NUMBER_OF_LONGS * Long.BYTES);
 
-        long l0 = toUnsignedLong(getInt(left, 0));
-        long l1 = toUnsignedLong(getInt(left, 1));
-        long l2 = toUnsignedLong(getInt(left, 2));
-        int l3raw = getRawInt(left, 3);
-        boolean leftNegative = isNegative(l3raw);
-        long l3 = toUnsignedLong(unpackUnsignedInt(l3raw));
+        multiply(getRawLong(left, 0), getRawLong(left, 1), getRawLong(right, 0), getRawLong(right, 1), result);
```

Comment:
`toUnsignedLong(getInt(left, 0))` is different than `getRawLong(left, 0)`.

One expect 32 bits storage space, while other expects 64 bits.

How does it work?

Reply:
toUnsignedLong(getInt(left, 0)) does:
- get low 32-bits from left. If we replace 0 with 3 then the high-most bit (sign) is erased
- cast them to long as unsigned

while getRawLong(left, 0) gets low 64 bits without any logic.
This is by no mean a replacement for the above. The actual sign logic and making 32-bit values is done inside the multiply method